### PR TITLE
feat: add featured image upload support

### PR DESCRIPTION
**Pull request by avnerI**

### CODE REVIEW

File: src/pages/api/complaint/create.ts [hunk 1]
```diff
@@ -100,10 +100,15 @@ export default async function handler(req: ExtendedNextApiRequest, res: NextApiR
     res.status(400).send('No files were uploaded.');
     return;
   }
-
+  debugger;
```
Comment (lisarisrael) [inline src/pages/api/complaint/create.ts:103]: debugger

File: src/pages/feed/index.tsx [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+import dayjs from 'dayjs';
+import Image from 'next/image';
 import Link from 'next/link';
 
 const FeedImage: React.FC<{ url: string; id: string }> = ({ url, id }) => {
   return (
     <div>
       <Link href={`/complaint/${id}`}>
-        <img
+        <Image
```
Comment (lisarisrael) [inline src/pages/feed/index.tsx:11]: like

File: src/pages/complaint/create.tsx [hunk 3]
```diff
@@ -187,6 +197,30 @@ const NewPost = () => {
...
+                        }}
+                      >
+                        <IconButton
+                          onClick={() => updateFeaturedImage(index)}
+                          aria-label="delete"
+                          style={{ opacity: 1 }}
+                        >
+                          <StarIcon
+                            style={{
+                              color: featuredImage === image.id || (!featuredImage && index === 0) ? 'yellow' : 'gray',
```
Comment (lisarisrael) [inline src/pages/complaint/create.tsx:219]: move colors to const

File: src/pages/feed/index.tsx [hunk 4]
```diff
@@ -30,19 +36,20 @@ const Feed: React.FC<{ complaints: Complaint[] }> = ({ complaints }) => {
             display: 'flex',
           }}
         >
-          <div>
+          <div style={{ padding: '24px 36px', border: 'solid', borderRadius: '4px', alignItems: 'flex-end' }}>
             {complaint.images.length > 0 && <FeedImage url={complaint.images[0]} id={complaint.complaintId} />}
```
Comment (lisarisrael) [inline src/pages/feed/index.tsx:40]: we don't really use the featured image

**Review by lisarisrael — CHANGES_REQUESTED** [review]

File: src/pages/feed/index.tsx [hunk 5]
```diff
@@ -51,7 +58,7 @@ export default Feed;
 
 export async function getServerSideProps() {
   const prisma = new PrismaClient();
-  const complaints = await prisma.complaint.findMany();
+  const complaints = await prisma.complaint.findMany({ orderBy: { createdAt: 'desc' } });
```
Comment (lisarisrael) [inline src/pages/feed/index.tsx:61]: this is part of the feature?

File: src/pages/complaint/create.tsx [hunk 6]
```diff
@@ -159,7 +169,7 @@ const NewPost = () => {
               {images &&
                 images.length > 0 &&
                 images.map((image, index) => (
-                  <div key={`${image.name}`} style={{ display: 'flex', alignItems: 'center', margin: '16px 0' }}>
+                  <div key={`${image.id}`} style={{ display: 'flex', alignItems: 'center', margin: '16px 0' }}>
```
Comment (lisarisrael) [inline src/pages/complaint/create.tsx:172]: nice

File: src/types/complaints/create.ts [hunk 7]
```diff
@@ -1,7 +1,8 @@
 export interface IFormInput {
   title: string;
   content: string;
-  images: File[];
+  images: { file: File; id: string }[];
```
Comment (lisarisrael) [inline src/types/complaints/create.ts:4]: this is not really letting us access the images by id, change it to Ditcenery?

**Review by lisarisrael — CHANGES_REQUESTED** [review]

File: src/pages/api/complaint/create.ts (hunk 1 above)
Comment (avnerI) [reply src/pages/api/complaint/create.ts]: Will add lintter soon in pre-push

File: src/types/complaints/create.ts (hunk 7 above)
Comment (avnerI) [reply src/types/complaints/create.ts:4]: will change in a flow up

File: src/pages/feed/index.tsx (hunk 5 above)
Comment (avnerI) [reply src/pages/feed/index.tsx:61]: just a nice to have

File: src/pages/complaint/create.tsx (hunk 3 above)
Comment (avnerI) [reply src/pages/complaint/create.tsx:219]: soon

**Review by lisarisrael — APPROVED** [review]

**Review by lisarisrael — APPROVED** [review]

**Pull request merged**